### PR TITLE
Implemented hooks

### PR DESCRIPTION
**Pull request by gabrielyea**
# Added the following:
- Change calculator to function.
- Change display and buttons to stateless functions.
- All buttons are generated from the same method to keep code DRY
- Added try-catch for zero divisions.
- Added small delegate pattern to change event handler on runtime (to dispatch error).
- [Live Demo](https://gabrielyea.github.io/math-magicians/)

### CODE REVIEW

File: src/components/calculator/Calculator.js [hunk 1]
```diff
@@ -1,101 +1,83 @@
...
+      <ul className="numbers-container">
+        {createButtons(numberKeys)}
+        {createButtons(decimalKey)}
+      </ul>
+      <ul className="operator-container">
+        {createButtons(operatorKeys)}
+      </ul>
+    </div>
+  );
 }
```
Comment (arslanbisharat) [inline src/components/calculator/Calculator.js]: - [x] While you have done a great job with the project and change class to function but I would recommend you to use ES6 sntax. Please work on this change.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [x]  Please remove .vscode by adding it to .gitgnore. This will help you make your project directory clean. 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/calculator/Calculator.js (hunk 1 above)
Comment (gabrielyea) [reply src/components/calculator/Calculator.js]: Done

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @gabrielyea, 

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**